### PR TITLE
Add timeout handling

### PR DESCRIPTION
**Pull request by ShimmerGlass**
* Add iddle timeout handling. The spec requires that SPOE server cleanly
close connections before haproxy's `timeout idle` configured value
* Add read and write timeout

The default values for timeouts are :
* iddle: 30s
* read: 1s
* write: 1s

### CODE REVIEW

File: disconnect.go [hunk 1]
```diff
@@ -1,25 +1,53 @@
...
+		}
 
-	message, _ := data["message"].(string)
-	if message == "" {
-		message = "unknown error "
+		log.Errorf("disconnect error: %s", message)
 	}
 
-	return fmt.Errorf("disconnect error: %s", message)
+	return nil
```
Comment (pierresouchay) [inline disconnect.go:53]: why is the error not returned anymore?

File: disconnect.go [hunk 2]
```diff
@@ -1,25 +1,53 @@
...
+	if !ok || code != 0 {
+		message, _ := data["message"].(string)
+		if message == "" {
+			message = "unknown error "
+		}
 
-	message, _ := data["message"].(string)
-	if message == "" {
-		message = "unknown error "
+		log.Errorf("disconnect error: %s", message)
```
Comment (pierresouchay) [inline disconnect.go]: why don't we return this instead of logging?

**Review by pierresouchay — CHANGES_REQUESTED** [review]
Looks good globally, but I wonder why we don't return error anymore in the fuction

**Comment by ShimmerGlass** [comment]
@pierresouchay I changed it back to return the error. It was a mistake.

**Pull request merged**